File: solution/tenkei/tenkei-38.py
def gcd(x, y):
    a = max(x, y)
    b = min(x, y)
    r = a % b
    if r == 0:
        return b
    return gcd(b, r)

a, b = map(int, input().split())

d = gcd(a, b)
p = a // d
q = b

if p * q > 1000000000000000000:
    print("Large")
    exit()

print(int(p * q))

File: solution/tenkei/tenkei-38.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;

ll gcd (ll x, ll y) {
  ll a = max(x, y);
  ll b = min(x, y);

  if (a % b == 0) return b;

  ll r = a % b;
  return gcd(b, r);
}

int main() {
  ll a, b;
  cin >> a >> b;

  ll d = gcd(a, b);

  ll p = a / d;
  ll q = b;

  if (p <= ll(1e18) / q) {
    cout << p * q << endl;
    return 0;
  }

  cout << "Large" << endl;
}
